llm_toolkit/code_fixer.py

```diff
@@ -656,6 +657,28 @@ def _collect_instruction_extern(benchmark: benchmarklib.Benchmark) -> str:
   return instruction
 
 
+def _collect_consume_buffers(fuzz_target_source_code: str) -> str:
+  """Provides advice on the use of ConsumeBytes and ConsumeData"""
+
+  instruction = ''
+
+  for buffer_method in ['ConsumeBytes', 'ConsumeData']:
+    if buffer_method in fuzz_target_source_code:
+      instruction += (
+          'IMPORTANT: the harness source code contains a call to '
+          f'`{buffer_method}`. Whenever this function is used, you MUST validate'
+          ' the size of the vector returned, and make sure that the size of the '
+          f'vector is equal to argument given to `{buffer_method}`. If it is '
+          'not equal, the harness should not proceed.\n')
+      instruction += (
+          f'Furthermore, consider changing {buffer_method} to '
+          '`ConsumeRandomLengthString` for creating `char` buffers or strings. '
+          'In most cases, `ConsumeRandomLengthString` is preferred, and '
+          f'should be used instead of {buffer_method}\n')
```

Comment:
Nit: Do we need two instructions if the source code contains both `ConsumeBytes` and `ConsumeData`?
As an alternative to shorten the prompt, we can merge them to
```
IMPORTANT: the harness source code contains a call to `ConsumeBytes ` or `ConsumeBytes`. ...
```
if the code contains either.